fastify.js

```diff
@@ -226,45 +222,23 @@ function build (options) {
     req.id = genReqId(req)
     req.log = res.log = log.child({ reqId: req.id, level: context.logLevel })
     req.originalUrl = req.url
-    res._startTime = hasLogger ? now() : undefined
-    res._context = context
 
     req.log.info({ req }, 'incoming request')
 
-    if (hasLogger === true || context.onResponse !== null) {
-      res.on('finish', onResFinished)
-      res.on('error', onResFinished)
-    }
+    var request = new context.Request(params, req, urlUtil.parse(req.url, true).query, req.headers, req.log)
+    var reply = new context.Reply(res, context, request, res.log)
+
+    reply._setup(hasLogger)
```

Comment:
I’m not fond of this. Can you put it behind a symbol, or make it a top level function (whichever is faster).

Reply:
Done!